client.go

```diff
@@ -170,3 +171,10 @@ func onConnectHandler(client PubSub, o *clientOptions) mqtt.OnConnectHandler {
 		}
 	}
 }
+
+func defaultNewClientFunc() *atomic.Value {
```

Comment:
Also is this Go's idiomatic way for initializing package level variables?

Reply:
> Store() method results in panic if expected type is not provided

Correct, however, we have a single Store call only and we store a function in it. This should be fine.

> Is this Go's idiomatic way for initializing package level variables

Not necessarily, it is handy when global vars are initialised/modified right before program start. Here it is helping in avoiding mutex locks since we only need to change the value in tests or on program start, mutex are better suited for use-cases where read/write has to be synchronised at runtime many times. Here, we only perform 1 store op at init and then simply load the value when required.